channelserver/src/main.rs

```diff
@@ -120,6 +121,17 @@ fn main() {
     let server = Arbiter::start(|_| server::ChannelServer::default());
     let log = Arbiter::start(|_| logging::MozLogger::default());
     let msettings = settings.clone();
+    let mut allowlist: Vec<ipnet::IpNet> = Vec::new();
+    // Add the list of known proxies.
+    if settings.known_proxy_list.len() > 0 {
+        for mut proxy in settings.known_proxy_list.split(",") {
+            proxy = proxy.trim();
+            if proxy.len() > 0 {
+                let addr:ipnet::IpNet = proxy.parse().unwrap();
```

Comment:
Using CIDRs here feels right to me, but part of me wonders whether we risk making them too broad in practice.  For example, if we were to declare that the entire AWS ELB internal IP range was "trusted", would this would allow an attacker to stand up their own ELB, somehow configure it to send fake X-Forwarded-For headers, and bypass our security checks by routing their traffic through it?

An alternative approach would be to do similar to what FxA does, where we basically have a config option to declare "we expect N well-behaved trusted proxies in front of the app", and then just take the Nth rightmost item from X-Forwarded-For as the ultimate client IP address:

  https://github.com/mozilla/fxa-auth-server/blob/master/lib/server.js#L198

For FxA in production the value of N is two - one for nginx, and one for the ELB.

Reply:
It's kind of a hard thing, I'll agree. We would need to be reasonably careful about what sort of sites we add, but if someone is coming at us from an AWS site, I'm not so sure we shouldn't block that. We're supposed to be accepting from clients, which generally aren't run out of AWS blocks.